src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -135,6 +144,34 @@
      */
     private EntityAIStatePausedHandler pausedHandler;
 
+    /*
+     * Setup all the interaction trigger predicates.
+     */
+    static
+    {
+        InteractionValidatorPredicates.addStandardInteractionValidatorPredicate(new TranslationTextComponent(COM_MINECOLONIES_COREMOD_ENTITY_WORKER_INVENTORYFULLCHEST), citizen -> citizen.getWorkBuilding() != null && InventoryUtils.isProviderFull(citizen.getWorkBuilding()));
+        InteractionValidatorPredicates.addPosBasedInteractionValidatorPredicate(
+          new TranslationTextComponent(BUILDING_LEVEL_TOO_LOW), (citizen, pos) ->
+          {
+              final IBuildingWorker workBuilding = citizen.getWorkBuilding();
+              if (workBuilding != null)
+              {
+                    final IColony colony = citizen.getColony();
+                    if ( colony != null )
+                    {
+                        final World world = colony.getWorld();
+                        if ( world != null )
+                        {
+                            return workBuilding.getMaxToolLevel() < WorkerUtil.getCorrectHavestLevelForBlock(world.getBlockState(pos).getBlock());
+                        }
+                    }
+              }
+              return false;
+          });
+        InteractionValidatorPredicates.addTokenBasedInteractionValidatorPredicate(new TranslationTextComponent(NORMAL_REQUEST),
+          (citizen, token) -> citizen.getColony() != null && ( citizen.getColony().getRequestManager().getPlayerResolver().holdsRequest(token) || citizen.getColony().getRequestManager().getRetryingRequestResolver().holdsRequest(token)));
```

Comment:
You can simply ask the manager which resolver is assigned and do an instance of check!

Reply:
kk